### PR TITLE
Adapt to new safe_ldb_search function

### PR DESCRIPTION
**Pull request by exekias**
Adapt to new OpenChange safe method to avoid freeing
connInfo->samCtx without replacing it with a new one on
reconnect events

### CODE REVIEW

**Comment by exekias** [comment]
This branch requires https://github.com/zentyal/openchange/pull/180 to work

File: OpenChange/MAPIStoreSamDBUtils.m [hunk 1]
```diff
@@ -29,8 +29,16 @@
 
 #import "MAPIStoreSamDBUtils.h"
 
+
+/* safe_ldb_search signature */
+int safe_ldb_search(struct ldb_context *ldb, TALLOC_CTX *mem_ctx,
+		    struct ldb_result **result, struct ldb_dn *base,
+		    enum ldb_scope scope, const char * const *attrs,
+		    const char *exp_fmt, ...);
```
Comment (sixstone-qq) [inline OpenChange/MAPIStoreSamDBUtils.m]: I'd rather prefer to publish the function in `libmapiproxy.h` as it is safer and more clear. Although this requires changes in OpenChange PR.

**Comment by exekias** [comment]
Updated, along with this openchange commit https://github.com/zentyal/openchange/commit/7bf57b03977f7c0c48738474d8ca0b4a49dcf899

**Comment by blaxter** [comment]
I don't like to add the definition to the public header (it's an utility function, nothing related with libmapiproxy) but I can live with that.

**Comment by blaxter** [comment]
retest this please

**Comment by blaxter** [comment]
This will require latest zentyal/openchange so we need to update the dependencies when building a new package

**Comment by infosite** [comment]
OpenChange Server.4.2.3
Zentyal 4.2.1.2

but I can not find the reposity to apply


2015-11-25 12:12 GMT+01:00 Jesús García Sáez <notifications@github.com>:

**Comment by infosite** [comment]
error git
fatal: repository '
https://github.com/zentyal/sogo/blob/jag/rtf-parser-fix-unicode/OpenChange/RTFHandler.m/'
not found

2015-11-25 12:19 GMT+01:00 Fabio Arbau <fabio.arbau@gmail.com>:

**Comment by blaxter** [comment]
@infosite What? I don't know what you are trying to do. That git command that you've tried to execute doesn't make any sense.

My guess is that you want to apply this change on your zentyal 4.2, that's not as simple as zentyal code (which is perl) because you have to compile the code and build the package. This branch (which is already merged) will be publish for zentyal 4.2 probably tomorrow (new package of both openchange and sogo). If you want to keep updated with changes check the release section of this repo [0] and zentyal/openchange [1] it's updated frequently.

[0] https://github.com/zentyal/sogo/releases
[1] https://github.com/zentyal/openchange/releases

**Comment by blaxter** [comment]
Ah, and this pull request has not relationship at all with RTF, the rtf pull request has to be reviewed and won't be neither merged nor released in short term.

**Pull request merged**